### PR TITLE
Refactor dataset importer

### PR DESCRIPTION
**Pull request by laurentqro**
Mostly extracted local variable to a method, removed arguments where we could use the available `dataset` reader method and refactored for clarity.

### CODE REVIEW

File: app/services/legacy/dataset_import_service.rb [hunk 1]
```diff
@@ -11,13 +11,14 @@ def initialize(legacy_dataset, orgs_cache, themes_cache)
   def run
     update_or_create_dataset
     create_inspire_dataset(dataset.id) if dataset.dataset_type == 'inspire'
-    create_resources(dataset)
+    create_datafiles
+    create_documents
   end
 
   def update_or_create_dataset
```
Comment (TobyRet) [inline app/services/legacy/dataset_import_service.rb]: @laurentqro (def) update_or_create_dataset is declared twice(?)

File: app/services/legacy/dataset_import_service.rb [hunk 2]
```diff
@@ -36,47 +37,76 @@ def update_or_create_dataset
...
+    dataset.theme_id = build_theme_id
+    dataset.secondary_theme_id = build_secondary_theme_id
     dataset.status = "published"
     dataset.save!(validate: false)
   end
 
-  def create_resources(dataset)
-    create_datafiles(dataset)
-    create_documents(dataset)
+  def update_or_create_dataset
```
Comment (TobyRet) [inline app/services/legacy/dataset_import_service.rb]: already declared?

**Comment by TobyRet** [comment]
I'm not clear on how the changes are speeding up the import process ... I must be missing something!

File: app/services/legacy/dataset_import_service.rb [hunk 3]
```diff
@@ -36,47 +37,76 @@ def update_or_create_dataset
...
+    if dataset.persisted?
+      dataset.update_columns(dataset_attributes)
```
Comment (TobyRet) [inline app/services/legacy/dataset_import_service.rb]: ... ah ... I think this is what is speeding up the process? It's making a direct call to the db

File: app/services/legacy/dataset_import_service.rb (hunk 1 above)
Comment (laurentqro) [reply app/services/legacy/dataset_import_service.rb]: good catch, fixed

**Pull request merged**